components/log-viewer-webui/server/src/S3Manager.js

```diff
@@ -0,0 +1,44 @@
+import {
+    GetObjectCommand,
+    S3Client,
+} from "@aws-sdk/client-s3";
+import {getSignedUrl} from "@aws-sdk/s3-request-presigner";
+
+
+/**
+ * Class for validating s3 inputs.
+ */
+class S3Manager {
+    #s3Client;
+
+    /**
+     * Creates S3 Manager.
+     *
+     * @param {string} region
+     */
```

Comment:
high level descriptions can be omitted for constructors unless strictly needed
```suggestion
    /**
     * @param {string} region
     */
```